metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -1045,9 +1049,19 @@ def get_dataset_properties() -> DatasetPropertiesClass:
             )
 
         def get_s3_tags() -> Optional[GlobalTagsClass]:
-            bucket_name = s3_util.get_bucket_name(
-                table["StorageDescriptor"]["Location"]
-            )
+            # try/catch bucket name fetch because when TableType=VIRTUAL_VIEW the Location can be empty
+            # and we should be able to return no tags rather than fail the entire ingestion
+            try:
+                bucket_name = s3_util.get_bucket_name(
+                    table["StorageDescriptor"]["Location"]
```

Comment:
It seems to me that we just need to be more defensive about this. When a StorageDescriptor -> Location entry is not found, don't try to get tags for that bucket. Shouldn't be raising exception either. 

So something like:
```python
if table.get("StorageDescriptor",{}).get("Location"):
   bucket_name = s3_util.get_bucket_name(table["StorageDescriptor"]["Location"])...
...
else:
   return None
```

might be enough.